### PR TITLE
feat(cloudfront): fromWebDistributionArn import method

### LINKED ISSUE #3606: CloudFrontWebDistribution external import method missing
CloudFrontWebDistribution currently does not have an external import method like S3 and Repository.

my thought is that it should support something like:
`cloudfront.CloudFrontWebDistribution.from_distribution_arn(self,<ID>,<ARN>)`

### PR DESCRIPTION
**Pull request by nmussy**
* add `CloudFrontWebDistributionBase` Resource class

Fixes #3606

----

Please read the [contribution guidelines](https://github.com/aws/aws-cdk/blob/master/CONTRIBUTING.md) and follow the pull-request checklist.

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts [hunk 1]
```diff
@@ -511,7 +517,30 @@ interface BehaviorWithOrigin extends Behavior {
...
+   */
+  public static fromWebDistributionArn(scope: cdk.Construct, id: string, distributionArn: string): IDistribution {
+    const stack = cdk.Stack.of(scope);
+    /*const region = stack.region;
+    const urlSuffix = stack.urlSuffix;*/
+
+    class Import extends CloudFrontWebDistributionBase {
+      public readonly distributionId = stack.parseArn(distributionArn).resourceName!;
+      // TODO
+      public readonly domainName = "";
```
Comment (nmussy) [inline packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts]: I'm not sure how to retrieve the `domainName` property, or any of the `CloudFrontWebDistributionProps`. 
Should they be manually by the user in their stack, or should I do something similar to `Vpc.fromLookup`?

Comment (eladb) [reply packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts]: This implies that `domainName` does not belong to `IDistribution`

Comment (nmussy) [reply packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts]: Fair enough. So we `@deprecate` it before removing it in v2?

And concerning the other props, we just leave them empty? Won't that cause issues during deployment if another resource tries to access one of them?

Comment (eladb) [reply packages/@aws-cdk/aws-cloudfront/lib/web_distribution.ts]: It’s an experimental package. You can break...

**Comment by shivlaks** [comment]
is this still being worked on? I can pick it up if needed...

**Comment by nmussy** [comment]
@shivlaks Sorry, I haven't looked into it further. Feel free to pick it up though, thanks!

**Comment by shivlaks** [comment]
closing this review as it's stale. it's better implemented in a fresh PR

**Pull request closed without merging**